ako-infra/ingestion/avi_infra_controller.go

```diff
@@ -102,165 +106,231 @@ func (a *AviControllerInfra) DeriveCloudNameAndSEGroupTmpl(tz string) (error, st
-			return false
+		if *seg.Name == clusterName {
+			return nil, &seg
 		}
 	}
-	return true
-}
-
-// ConfigureSeGroup creates the SE group with the supplied properties, alters just the SE group name and the labels.
-func ConfigureSeGroup(client *clients.AviClient, seGroup *models.ServiceEngineGroup) bool {
-	// Change the name of the SE group
-	*seGroup.Name = lib.GetClusterID()
-
-	uri := "/api/serviceenginegroup/"
-	// Add the labels.
-	seGroup.Labels = lib.GetLabels()
-	response := models.ServiceEngineGroupAPIResponse{}
-	SetAdminTenant := session.SetTenant(lib.GetAdminTenant())
-	SetTenant := session.SetTenant(lib.GetTenant())
-
-	err := lib.AviPost(client, uri, seGroup, response)
-	if err != nil {
-		if aviError, ok := err.(session.AviError); ok && aviError.HttpStatusCode == 403 {
-			//SE in provider context
-			utils.AviLog.Debugf("Switching to admin context from  %s", lib.GetTenant())
-			SetAdminTenant(client.AviSession)
-			defer SetTenant(client.AviSession)
-			err := lib.AviPost(client, uri, seGroup, response)
-			if err != nil {
-				utils.AviLog.Warnf("Error during POST call to create the SE group :%v", err.Error())
-				return false
 
-			}
-		} else {
-			utils.AviLog.Warnf("Error during POST call to create the SE group :%v", err.Error())
-			return false
+	if result.Next != "" {
+		// It has a next page, let's recursively call the same method.
+		next_uri := strings.Split(result.Next, "/api/serviceenginegroup")
+		if len(next_uri) > 1 {
+			overrideUri := "/api/serviceenginegroup" + next_uri[1]
+			nextPage := lib.NextPage{NextURI: overrideUri}
+			return fetchSEGroup(client, nextPage)
 		}
 	}
-	utils.AviLog.Infof("labels: %v set on Service Engine Group :%v", utils.Stringify(lib.GetLabels()), *seGroup.Name)
-	return true
 
+	utils.AviLog.Infof("No Service Engine Group found for Cluster.")
+	return nil, nil
 }
 
-func checkSeGroup(client *clients.AviClient, cloudName string) bool {
-	segroupName := lib.GetClusterID()
+func fetchVcenterServer(client *clients.AviClient) (string, error) {
+	uri := "/api/vcenterserver"
+	result, err := lib.AviGetCollectionRaw(client, uri)
+	if err != nil {
+		return "", err
+	}
+	if result.Count == 0 {
+		return "", fmt.Errorf("vcenterServer object not found")
+	}
+	elems := make([]json.RawMessage, result.Count)
+	err = json.Unmarshal(result.Results, &elems)
+	if err != nil {
+		return "", err
+	}
+	vc := avimodels.VCenterServer{}
+	err = json.Unmarshal(elems[0], &vc)
+	if err != nil {
+		return "", err
+	}
+	return *vc.Name, nil
+}
 
-	uri := "/api/serviceenginegroup/?name=" + segroupName + "&cloud_ref.name=" + cloudName
-	response := models.ServiceEngineGroupAPIResponse{}
-	err := lib.AviGet(client, uri, &response)
+// ConfigureSeGroup creates the SE group with the supplied properties, alters just the SE group name and the markers.
+func ConfigureSeGroup(client *clients.AviClient, seGroup *models.ServiceEngineGroup, segExists bool) bool {
+	// Change the name of the SE group, and add markers
```

Comment:
we can move line 286 to 293 after 298 as `fetchVcenter` fails, we will waste cycle doing this.